flexmeasures/ui/utils/view_utils.py

```diff
@@ -32,12 +33,21 @@ def render_flexmeasures_template(html_filename: str, **variables):
         variables["documentation_exists"] = True
 
     # use event_starts_after and event_ends_before from session if not given
-    variables["event_starts_after"] = variables.get(
-        "event_starts_after", session.get("event_starts_after")
-    )
-    variables["event_ends_before"] = variables.get(
-        "event_ends_before", session.get("event_ends_before")
+    variables["event_starts_after"] = session.get(
+        "event_starts_after"
+    ) or variables.get("event_starts_after")
+    variables["event_ends_before"] = variables.get("event_ends_before") or session.get(
+        "event_ends_before"
     )
+    if variables["event_starts_after"]:
+        variables["event_starts_after"] = (
+            datetime.fromisoformat(variables["event_starts_after"]) + timedelta(days=1)
+        ).isoformat()
+
+    if variables["event_ends_before"]:
+        variables["event_ends_before"] = (
+            datetime.fromisoformat(variables["event_ends_before"]) - timedelta(days=1)
```

Comment:
Please explain why we subtract a day here, and add one above. Like this, it is confusing. We're creating more code that is difficult to debug.

I actually didn't remember why we have to do this. It seems due to timezone not being transferred correctly. Felix [wrote](https://github.com/FlexMeasures/flexmeasures/pull/1165#pullrequestreview-2286191779) this: "This seems to be due to the timezone of `event_starts_after` being set to UTC somewhere." So then we move from 00:00:00 o'clock to 23:00:00. How is adding 24 hours fixing this?

Did you test this well with different ranges - like 1 day, 2 days, 5 days?

Reply:
This was my attempt at trying different approaches after Felix's initial suggestion didn’t completely resolve the issue. I thought handling the deduction on the backend might work better, and while it did, it introduced unnecessary complexity to the code.

I've since removed those changes in view_utils and returned to base.html. I retried Felix’s suggestion, adding an extra step that finally fixed the issue:
```
 var start_time = '{{event_starts_after}}';
 var new_start_date = new Date(start_time);
 toIsoString(new_start_date)
```

Without this additional step, the start date kept shifting back by one day, which initially led me to believe the issue was elsewhere. Here’s a log showing the timezone difference:

```
start_time:  2024-09-09T23:00:00.000Z
new_start_date:  Tue Sep 10 2024 00:00:00 GMT+0100 (West Africa Standard Time)
new_start_date tostring:  2024-09-10T00:00:00+01:00
```